tests/bin/sessions.test.ts

```diff
@@ -1,215 +1,321 @@
+  });
+  test('failing processes should unlock the session file', async () => {
+    const session = await Session.createSession({
+      sessionTokenPath,
+      fs,
+      logger,
     });
-
-    test(
-      'Failing processes should unlock the session file',
-      async () => {
-        const result = await testUtils.pkStdio(
-          ['vaults', 'delete', 'NotAVault', '-np', nodePath],
-          {},
-          dataDir,
-        );
-        expect(result.exitCode).not.toBe(0);
-
-        tokenBuffer = await fs.promises.readFile(sessionFile);
-        token = tokenBuffer.toString() as SessionToken;
-        expect(
-          async () => await polykeyAgent.sessionManager.verifyToken(token),
-        ).not.toThrow();
-
-        // Try to lock the session file to ensure it's unlocked
-        const fd = fs.openSync(sessionFile, 'r');
-        expect(lock(fd)).toBeTruthy();
-        lock.unlock(fd);
-        fs.closeSync(fd);
+    // Run command that will fail
+    const { exitCode } = await testBinUtils.pkStdio(
+      ['identities', 'search', 'InvalidProvider'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
       },
-      global.failedConnectionTimeout,
+      global.binAgentDir,
     );
-
-    test('Parallel processes should not refresh the session token', async () => {
-      let tokenP1 = 'token1' as SessionToken;
-      let tokenP2 = 'token2' as SessionToken;
-
-      tokenBuffer = await fs.promises.readFile(sessionFile);
-      const prevTokenParallel = tokenBuffer.toString() as SessionToken;
-
-      async function runListCommand(): Promise<SessionToken> {
-        // At least 1 second of delay
-        // Expiry time resolution is in seconds
-        await sleep(1000);
-        await testUtils.pkStdio(command, {}, dataDir);
-        const buffer = await fs.promises.readFile(sessionFile);
-        return buffer.toString() as SessionToken;
-      }
-
-      [tokenP1, tokenP2] = await Promise.all([
-        runListCommand(),
-        runListCommand(),
-      ]);
-
-      // Verify both tokens
-      expect(
-        async () => await polykeyAgent.sessionManager.verifyToken(tokenP1),
-      ).not.toThrow();
-      expect(
-        async () => await polykeyAgent.sessionManager.verifyToken(tokenP2),
-      ).not.toThrow();
-
-      // Check that both commands were completed
-      expect(tokenP1).not.toEqual('token1');
-      expect(tokenP2).not.toEqual('token2');
-
-      // Check that the session token was refreshed exactly one time
-      if (tokenP1 === prevTokenParallel) {
-        expect(tokenP2).not.toEqual(prevTokenParallel);
-      } else if (tokenP2 === prevTokenParallel) {
-        expect(tokenP1).not.toEqual(prevTokenParallel);
-      } else {
-        expect(tokenP1).toEqual(tokenP2);
-      }
+    expect(exitCode).not.toBe(0);
+    await sleep(1100);
+    // Write to session file to ensure it's unlocked
+    await session.writeToken('abc' as SessionToken);
```

Comment:
Should also add an attempt to do `writeToken` at the beginning of this test, to assert that the session file is initially locked.

Reply:
Might be misunderstanding `session` here and the test description, but I'm able to `writeFile` before the command is run with no issues (i.e. the session file is unlocked even before the command is run):

```ts
  test.only('failing processes should unlock the session file', async () => {
    const session = await Session.createSession({
      sessionTokenPath,
      fs,
      logger,
    });
    await session.writeToken('newToken' as SessionToken); // this works - and the session token becomes "newToken")
    // Run command that will fail
    const { exitCode } = await testBinUtils.pkStdio(
      ['identities', 'search', 'InvalidProvider'],
      {
        PK_NODE_PATH: global.binAgentDir,
        PK_PASSWORD: global.binAgentPassword,
      },
      global.binAgentDir,
    );
    expect(exitCode).not.toBe(0);
    await sleep(1100);
    // Write to session file to ensure it's unlocked
    await session.writeToken('abc' as SessionToken);
    const token = await session.readToken();
    expect(token).toEqual('abc');
  });
```

Reply:
What do you mean `writeFile`? Yea well you can write some random crap into the token. A normal `fs.promises.writeFile` will work. The token should then be incorrect.

Reply:
Ah, I understand now. So after these locking behaviour changes, is the session file still locking during the process execution? If so, this test seems fine to me, and I'll just reword the description. However, if it isn't locked anymore, then we can remove the test entirely.

Reply:
From @emmacasolin :

> I don’t believe so - there are separate read and write locks now so the session file shouldn’t be locked for the entire duration of the command’s execution, only when the file is being read/written to.

In that case, my opinion is this test can be removed entirely.

Reply:
Done